### PR TITLE
optimizer-hints: add docs for `nth_plan()` hint

### PR DESCRIPTION
**Pull request by time-and-fate**
### What is changed, added or deleted? (Required)

Add dos for `nth_plan()` hint in `/optimizer-hints.md`.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- Other reference link(s): https://github.com/pingcap/tidb/issues/17580

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by TomShawn** [comment]
@lzmhhh123 @eurekaka Please also help confirm whether this PR applies to v4.0, thanks!

**Comment by SunRunAway** [comment]
> Should we expose this hint to users? I believe this hint is just for testing purpose, users may have no expectation about what the result plan of `nth_plan(xxx)` would be? @SunRunAway

I think it is ok to expose this to users, and we should state that it is for testing purposes and may not be compatible in the future.

**Review by SunRunAway — COMMENTED** [review]
LGTM

File: optimizer-hints.md [hunk 1]
```diff
@@ -313,3 +313,23 @@ SELECT /*+ READ_CONSISTENT_REPLICA() */ * FROM t;
 ```sql
 prepare stmt FROM 'SELECT  /*+ IGNORE_PLAN_CACHE() */ * FROM t WHERE t.id = ?';
 ```
+
+### NTH_PLAN(N)
+
+`NTH_PLAN(N)` 提示优化器选用在物理优化阶段搜索到的第 `N` 个物理计划。`N` 必须是正整数。
+
+如果指定的 `N` 超出了物理优化阶段的搜索范围，TiDB 会返回 warning，并使用和没有该 Hint 时一样的策略选择最优物理计划。
```
Comment (TomShawn) [inline optimizer-hints.md:321]:
```suggestion
如果指定的 `N` 超出了物理优化阶段的搜索范围，TiDB 会返回 warning，并根据不存在该 Hint 时一样的策略选择最优物理计划。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**